packages/atclient/include/atclient/atkey.h

```diff
@@ -51,6 +52,16 @@ void atclient_atkey_free(atclient_atkey *atkey);
  */
 int atclient_atkey_from_string(atclient_atkey *atkey, const char *atkeystr, const size_t atkeylen);
 
+/**
+ * @brief get the length of the atkey string
+ *
+ * @param atkey atkey struct to read, assumed that this was already initialized via atclient_atkey_init
+ * @return size_t the length of the atkey string
+ *
+ * @note this excludes the null terminator and metadata string fragement
+ */
+size_t atclient_atkey_strlen(const atclient_atkey *atkey);
+
```

Comment:
New function, currently **highly** unoptimized, but I am using it to keep the notify implementation clean.

Left a TODO in atkey.c which can be addressed later.